services/graphql-server/src/graphql/definitions/email/campaign.js

```diff
@@ -1,11 +1,80 @@
 const gql = require('graphql-tag');
 
 module.exports = gql`
+  extend type Query {
+    emailCampaign(input: EmailCampaignQueryInput!): EmailCampaign
+      @findOne(model: "email.Campaign", using: { id: "_id" })
+    emailCampaigns(input: EmailCampaignQueryInput!): EmailCampaignConnection
+  }
 
-type EmailCampaignTestRecipient {
-  firstName: String
-  lastName: String
-  email: String
-}
+  input EmailCampaignQueryInput {
+    status: ModelStatus = active
+    id: ObjectID
+    productId: ObjectID
+    sort: EmailCampaignsSortInput = {}
+    pagination: PaginationInput = {}
+    beginning: CampaignBeginningInput = {}
+    ending: CampaignEndingInput = {}
+  }
+
+  input CampaignBeginningInput {
+    before: Date
+    after: Date
+  }
+
+  input CampaignEndingInput {
+    before: Date
+    after: Date
+  }
+
+  enum EmailCampaignsSortField {
+    status
+    deploymentDate
+    scheduled
+  }
+
+  input EmailCampaignsSortInput {
+    field: EmailCampaignsSortField = deploymentDate
+    order: SortOrder = desc
+  }
+
+  type EmailCampaignConnection @projectUsing(type: "EmailCampaign") {
+    totalCount: Int!
+    edges: [EmailCampaignEdge]!
+    pageInfo: PageInfo!
+  }
+
+  type EmailCampaignEdge {
+    node: EmailCampaign!
+    cursor: String!
+  }
+
+  type EmailCampaign {
+    id: ObjectID! @projection(localField: "_id") @value(localField: "_id")
+    status: Int @projection
+    name: String @projection
+    created: Date @projection
+    touched: Date @projection
+    updated: Date @projection
+    createdBy: Date @projection
+    deploymentDate: Date @projection
+    scheduled: Date @projection
+    htmlDate: Date @projection
+    externalId: String @projection
+    fromName: String @projection
+    html: String @projection
+    listId: Int @projection
+    listStatus: String @projection
+    locked: String @projection
+    subjectLine: String @projection
+    listMessage: String @projection
+    tests: [EmailCampaignTestRecipient] @projection
```

Comment:
`tests` should be the type `EmailCampaignTest` not `EmailCampaignTestRecipient`. It should also always be an array, if you're planning to use this field.

Reply:
I don't plan on using this, it was in the file when I started. Should I just remove it?